File: BL/src/main/java/com/eighteen/muricabl/DTO/_Interfaces/IUser.java
package com.eighteen.muricabl.DTO._Interfaces;

import java.time.LocalDate;

/**
 * Interface to represent the user model
 */
public interface IUser {
    //region Getters
    String getId();
    String getName();
    String getEmail();
    String getPasswordHash();
    LocalDate getBirthDate();
    //endregion

    //region Setters
    IUser setId(String id);
    IUser setName(String name);
    IUser setEmail(String email);
    IUser setPasswordHash(String passwordHash);
    IUser setBirthDate(LocalDate birthDate);
    //endregion
}

File: murica_bl/DTO/_Interfaces/IUser.php
<?php

namespace murica_bl\DTO\_Interfaces;

interface IUser
{
    public function getId(): string;
    public function getName(): string;
    public function getEmail(): string;
    public function getPassword(): string;
    public function getBirthDate(): string;

    public function setId(string $id): IUser;
    public function setName(string $name): IUser;
    public function setEmail(string $email): IUser;
    public function setPassword(string $password): IUser;
    public function setBirthDate(string $birth_date): IUser;
}